xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_shader_submit_new.cpp

```diff
@@ -1,8 +1,329 @@
 #include "stdafx.h"
 #include "rasterizer_dx9_shader_submit_new.h"
 
+#include "rasterizer_dx9_main.h"
+#include "rasterizer_dx9_targets.h"
+
+#include "interface/new_hud_draw.h"
+#include "objects/light_definitions.h"
+#include "rasterizer/rasterizer_globals.h"
+#include "rasterizer/rasterizer_lights.h"
+#include "render/render_first_person.h"
+#include "shaders/shader_definitions.h"
+#include "shaders/shader_templates.h"
+
+/* globals */
+
+#ifndef NDEBUG
+bool g_spherical_light_error_displayed = false;
+#endif
+
+bool g_use_secondary_gradient_index = false;
+bool g_use_gel_map_bitmap = true;
+
+/* prototypes */
+
+void __fastcall c_shader_submission_interface_new__stage_texture(
+    c_shader_submission_interface_new* _this,
+    uint32 _edx,
+    int16 stage,
+    e_shader_pass_texture_source_extern bitmap_extern_index,
+    uint32* res_x,
+    uint32* res_y
+);
+
+/* public code */
+
+void rasterizer_dx9_shader_submit_new_apply_patches(void)
+{
+    PatchCall(Memory::GetAddress(0x269091), c_shader_submission_interface_new__stage_texture);
+    return;
+}
+
+void c_shader_submission_interface_new::stage_texture(
```

Comment:
what is this for?

Reply:
It's for fixing the particle distortion effect

Reply:
yeah, didn't notice the StretchRect